### PR TITLE
feat/proposal: Deprecate jaeger enabled

### PR DESCRIPTION
**Pull request by markphelps**
Fixes: FLI-197
Fixes: #1310 

Deprecates `tracing.jaeger.enabled` in favor of:

```yaml
tracing:
  enabled: true
  backend: jaeger
```

### CODE REVIEW

File: config/flipt.schema.cue [hunk 1]
```diff
@@ -129,6 +131,9 @@ import "strings"
 	}
 
 	#tracing: {
+		enabled?: bool | *false
+		exporter?: "jaeger" | *"jaeger"
```
Comment (GeorgeMac) [inline config/flipt.schema.cue:135]:
```suggestion
		enabled: bool | *false
		exporter: "jaeger" | *"jaeger"
```

I think you can forgo making this optional if there is a default.

Comment (GeorgeMac) [reply config/flipt.schema.cue:135]: Actually with `exporter` you might be able to get away with just:
```
exporter?: jaeger
```
Which just says if you specify the key `exporter` it must be the value `"jaeger"`.
And drop the default.

**Review by GeorgeMac — APPROVED** [review]
Nice. One observation that I am not 💯 is necessary.

**Comment by markphelps** [comment]
@GeorgeMac I'm wondering if we should change `exporter` to be `collector` or maybe something more agnostic like `backend` or `consumer` ? `backend` might be nice as it mimics what we have in the `cache` config as well.

Looking at [OTel's glossary](https://opentelemetry.io/docs/reference/specification/overview/), it seems to me that:

- [`exporter`](https://opentelemetry.io/docs/reference/specification/glossary/#exporter-library) is actually the library used to send the traces
- [`collector`](https://opentelemetry.io/docs/reference/specification/overview/#collector) seems specific to their OTLP implementation


WYT?

**Comment by GeorgeMac** [comment]
@markphelps I like your reasoning there. `backend` sounds good. Other ideas could be `destination` or `target`.
But I like `backend` just as much.

File: config/flipt.schema.cue (hunk 1 above)
Comment (markphelps) [reply config/flipt.schema.cue:135]: i'll prob just leave this as is for now since we will very shortly allow others besides `jaeger`

**Review by GeorgeMac — APPROVED** [review]
Sweet.

**Pull request merged**